src/linear_algebra/lagrange.lean

```diff
@@ -156,12 +179,54 @@ of degree less than `s.card`. -/
 def fun_equiv_degree_lt : degree_lt F s.card ≃ₗ[F] (s → F) :=
 { to_fun := λ f x, f.1.eval x,
   map_add' := λ f g, funext $ λ x, eval_add,
-  map_smul' := λ c f, funext $ λ x, by
-    { change eval ↑x (c • f).val = (c • λ (x : s), eval ↑x f.val) x,
-      rw [pi.smul_apply, smul_eq_mul, ← @eval_C F c _ x,
-      ← eval_mul, eval_C, C_mul'], refl },
-  inv_fun := λ f, ⟨interpolate s f, mem_degree_lt.2 $ degree_interpolate_lt s f⟩,
-  left_inv := λ f, subtype.eq $ eq_interpolate s f $ mem_degree_lt.1 f.2,
-  right_inv := λ f, funext $ λ ⟨x, hx⟩, eval_interpolate s f x hx }
+  map_smul' := λ c f, funext $ by simp,
+  inv_fun := λ f, ⟨interpolate s (λ x, if hx : x ∈ s then f ⟨x, hx⟩ else 0),
+    mem_degree_lt.2 $ degree_interpolate_lt _ _⟩,
+  left_inv := λ f, begin apply subtype.eq,
+    simp only [subtype.coe_mk, subtype.val_eq_coe, dite_eq_ite],
+    convert eq_interpolate s f (mem_degree_lt.1 f.2) using 1,
+    rw interpolate_eq_of_eval_eq,
+    intros x hx,
+    rw if_pos hx end,
+  right_inv := λ f, funext $ λ ⟨x, hx⟩, begin
+    convert eval_interpolate s _ x hx,
+    simp_rw dif_pos hx end }
+
+theorem interpolate_eq_interpolate_erase_add {x y : F} (hx : x ∈ s) (hy : y ∈ s) (hxy : x ≠ y) :
+  interpolate s f =
+  C (y - x)⁻¹ * ((X - C x) * interpolate (s.erase x) f + (C y - X) * interpolate (s.erase y) f) :=
+begin
+  apply eq_interpolate_of_eval_eq,
```

Comment:
You could change this into
```lean
refine eq_interpolate_of_eval_eq _ _ _ (λ z hz, _),
```
to save one line from `intros z hz`, though this is admittedly a pretty insignificant golf.